core/src/test/scala/unit/kafka/tools/ConsoleProducerTest.scala

```diff
@@ -128,19 +128,22 @@ class ConsoleProducerTest {
     assert(reader.parseKey)
   }
 
-  @Test(expected = classOf[IllegalArgumentException])
+  @Test
   def testBrokerListAndBootstrapServerOptionMissing(): Unit = {
-    val args: Array[String] = Array(
-      "--topic",
-      "producerTest"
-    )
-
-    CommandLineUtils.exitPolicy(new ExitPolicy {
-      override def exit(msg: String): Nothing = {
-        throw new IllegalArgumentException
-      }
-    })
-    new ConsoleProducer.ProducerConfig(args)
+    try {
```

Comment:
I can do this differently for example with var statuscode, var message variables and assert those in the end of the test. That way we don't need to deal with exceptions.